File: Python solutions/Arrays/two_sum.py
from ast import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
    
        #Initialize a map tp store a num and its index as key value pair
        
        map = {}
        idx = 0
        
        
        # Iterate over the all the nums in list and check if its complementary num
        # needed for two sum is present in map. If present return corresponding indexes.
        # Else add the current num and its index in map.
        
        for i in nums:
            key = target - i
            if key in map:
                return (idx,map[key])
            map[i] = idx
            idx += 1
        return (-1, -1)

File: Java solutions/Arrays/twoSum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {

        // Initialize a map to store the number encountered so far and its index.
        Map<Integer,Integer> map = new HashMap<>();

        //Iterate over all the number in the given array.
       for(int i = 0; i < nums.length; i++)
       {

           //key is the complement of the number you currently have.
           int key = target - nums[i];

           // If the complement is present in the map, you have your solution.
           // Else add the current number and its index in the map.
           if(map.containsKey(key))
           {
               return new int[]{i,map.get(key)};
           }
           else
           {
               map.put(nums[i],i);
           }
       }

       // If target is never found return -1 -1 to indicate target not found.
       return new int[]{-1, -1};
    }
}